Sources/xcproj/Referenceable.swift

```diff
@@ -0,0 +1,112 @@
+//
+//  Referenceable.swift
+//  xcprojPackageDescription
+//
+//  Created by rmalik on 11/8/17.
+//
+
+import Foundation
+
+// Typealias that represents dictionary from string -> T where T is Referenceable (i.e. PBXObjects)
+
+public typealias ReferenceableCache<T: Referenceable & Equatable> = [String: T]
```

Comment:
It's not just a cache it's our internal data store now. So maybe another name? `ReferenceCollection`?